tests/UnifiedSpecTests/RunOnRequirement.php

```diff
@@ -69,17 +113,7 @@ public function isSatisfied(string $serverVersion, string $topology, stdClass $s
             return false;
         }
 
-        if (isset($this->topologies)) {
-            if (in_array($topology, $this->topologies)) {
-                return true;
-            }
-
-            /* Ensure "sharded-replicaset" is also accepted for topologies that
-             * only include "sharded" (agnostic about the shard topology) */
-            if ($topology === self::TOPOLOGY_SHARDED_REPLICASET && in_array(self::TOPOLOGY_SHARDED, $this->topologies)) {
-                return true;
-            }
-
+        if (isset($this->topologies) && ! $this->isTopologySatisfied($topology)) {
```

Comment:
This fixes a subtle bug introduced by the versioned API work. When topology was the last thing checked, we could `return true`; however, that doesn't work if we need to do other assertions later on in this function.